bodo/pandas/physical/join.h

```diff
@@ -92,16 +92,53 @@ class PhysicalJoin : public PhysicalSourceSink, public PhysicalSink {
         std::shared_ptr<bodo::Schema> probe_table_schema_reordered =
             probe_table_schema->Project(probe_col_inds);
 
+        for (const duckdb::JoinCondition& cond : conditions) {
+            auto& left_bce =
+                cond.left->Cast<duckdb::BoundColumnRefExpression>();
+            auto& right_bce =
+                cond.right->Cast<duckdb::BoundColumnRefExpression>();
+
+            std::shared_ptr<PhysicalExpression> new_phys_expr =
+                std::static_pointer_cast<
+                    PhysicalExpression>(std::make_shared<
+                                        PhysicalComparisonExpression>(
+                    std::make_shared<PhysicalColumnRefExpression>(
+                        this->probe_col_inds_rev[getColRefMap(
+                            left_bindings)[{left_bce.binding.table_index,
+                                            left_bce.binding.column_index}]],
+                        left_bce.GetName(), true),
+                    std::make_shared<PhysicalColumnRefExpression>(
+                        this->build_col_inds_rev[getColRefMap(
+                            right_bindings)[{right_bce.binding.table_index,
+                                             right_bce.binding.column_index}]],
+                        right_bce.GetName(), false),
+                    cond.comparison));
+            if (physExprTree) {
+                physExprTree = std::static_pointer_cast<PhysicalExpression>(
```

Comment:
Add a check to make sure the data types are numeric and throw an error otherwise?

Technically we can support non-numerics like string too since we have table pointers (I think that's the reason for adding table points to the APIs). But can be follow up since it's more work.

Reply:
It is going to throw an error later somewhere else if the type isn't supported.  Is there a real advantage of doing it here?  I added some datetime stuff so we'd have to keep that set and any check here in sync which seems like too much work.